sematic/db/migrations/20230706150252_pre_alembic_parity.py

```diff
@@ -0,0 +1,339 @@
+# flake8: noqa E501
+# Sematic
+from sematic.db.db import db
+
+
+def up():
+    if db().get_engine().url.drivername == "sqlite":
+        with db().get_engine().begin() as conn:
+            conn.execute(
+                "UPDATE runs SET function_path = 'UNKNOWN.UNKNOWN' WHERE function_path IS NULL;"
+            )
+            conn.execute("UPDATE runs SET tags = '[]' WHERE tags IS NULL;")
+            conn.execute(
+                "UPDATE runs SET source_code = 'source code unavailable' WHERE source_code IS NULL;"
+            )
+            conn.execute(
+                "UPDATE jobs SET created_at = datetime(0, 'unixepoch') WHERE created_at IS NULL;"
+            )
+            conn.execute(
+                "UPDATE jobs SET updated_at = datetime(0, 'unixepoch') WHERE updated_at IS NULL;"
+            )
+            conn.execute(
+                "UPDATE notes SET created_at = datetime(0, 'unixepoch') WHERE created_at IS NULL;"
+            )
+            conn.execute(
+                "UPDATE notes SET updated_at = datetime(0, 'unixepoch') WHERE updated_at IS NULL;"
+            )
+
+            conn.execute(
+                """
+                CREATE TABLE runs_new (
+                    id character(32) NOT NULL,
+                    future_state TEXT NOT NULL,
+                    name TEXT,
+                    function_path TEXT NOT NULL,
+                    created_at timestamp WITH time zone NOT NULL,
+                    updated_at timestamp WITH time zone NOT NULL,
+                    started_at timestamp,
+                    ended_at timestamp,
+                    resolved_at timestamp,
+                    failed_at timestamp,
+                    parent_id character(32),
+                    description TEXT,
+                    tags TEXT NOT NULL,
+                    source_code TEXT NOT NULL,
+                    root_id character(32) NOT NULL,
+                    nested_future_id character(32),
+                    resource_requirements_json JSONB,
+                    exception_metadata_json JSONB,
+                    container_image_uri TEXT,
+                    external_exception_metadata_json JSONB,
+                    original_run_id character(32),
+                    cache_key TEXT,
+                    user_id character(32) REFERENCES users(id),
+
+                    PRIMARY KEY (id),
+
+                    FOREIGN KEY(root_id) REFERENCES runs (id)
+                );
+                """
+            )
+
+            conn.execute("CREATE INDEX ix_runs_cache_key ON runs_new (cache_key);")
```

Comment:
Doesn't `runs_cache_key_index` need to be removed? Or is it auto-removed when you drop the table? Does PostgreSQL behave differently? Same questions for all indexes.

Reply:
> Doesn't runs_cache_key_index need to be removed? Or is it auto-removed when you drop the table?

yes, indexes are removed when the table is dropped.

> Shouldn't you rather create the indexes after the rename anyway, for cleanliness?

done for all indexes.